activerecord/lib/active_record/fixtures.rb

```diff
@@ -743,11 +743,23 @@ def add_join_records(rows, row, association)
           column_type = association.primary_key_type
           lhs_key     = association.lhs_key
           rhs_key     = association.rhs_key
+          join_class  = table_name.classify.safe_constantize
 
           targets = targets.is_a?(Array) ? targets : targets.split(/\s*,\s*/)
+
           rows[table_name].concat targets.map { |target|
-            { lhs_key => row[primary_key_name],
-              rhs_key => ActiveRecord::FixtureSet.identify(target, column_type) }
+            join_row = {
+              lhs_key => row[primary_key_name],
+              rhs_key => ActiveRecord::FixtureSet.identify(target, column_type)
+            }
+            if join_class && join_class.record_timestamps
+              now = config.default_timezone == :utc ? Time.now.utc : Time.now
+              now = now.to_s(:db)
+              timestamp_column_names.each do |c_name|
```

Comment:
`timestamp_column_names` will be called on the fixtures class not on the join class. If the two have a different set of timestamp columns this breaks.